### PR TITLE
Prevent tserver shutdown when only 1 tserver running.

### LINKED ISSUE #1753: `admin stop` created deadlock in TabletServer being stopped
Taken from [ACCUMULO-3898](https://issues.apache.org/jira/browse/ACCUMULO-3898) and confirmed to still exist in accumulo. If you have multiple tservers and try using 'admin stop' to shutdown each individual tserver, the last tserver active will hang while trying to shut down. The issue on jira goes into more detail on the specificities happening during this process. 

I updated the IT that was attached to test this and that is on my fork here: 
[AdminStopTabletServersIT](https://github.com/Manno15/accumulo/blob/dcfb85a7a1096c238e3d694e49aa7ebd1f944c41/test/src/main/java/org/apache/accumulo/test/AdminStopTabletServersIT.java)

This IT sets up three tservers and when it tries to shut down the last one, will hang and the test will timeout. 

-Affected version(s) of this project: 2.0.0 for certain and a safe assumption that previous versions are affected as well.

### PR DESCRIPTION
**Pull request by milleruntime**
* Closes #1753
* Add new test to ShutdownTServerTest and rewrite single test

### CODE REVIEW

**Review by Manno15 — APPROVED** [review]
Looks good. Feel free to close my WIP PR #1770 when this gets merged.

**Review by ctubbsii — APPROVED** [review]
Looks good to me. The only thing I would suggest is to statically import EasyMock methods, just for slightly improved readability with the actual test code. It makes some lines shorter, and it's pretty easy to tell in a test where `createMock`, `expect`, `replay`, and `verify` are coming from.
But this is good to merge with or without that suggestion.

File: server/manager/src/main/java/org/apache/accumulo/manager/tserverOps/ShutdownTServer.java [hunk 1]
```diff
@@ -57,6 +57,13 @@ public long isReady(long tid, Manager manager) {
       return 0;
     }
 
+    // if one tserver is remaining throw an error and return before attempting shutdown
+    if (manager.onlineTabletServers().size() == 1) {
+      log.error("Only 1 tserver running and it contains the Root table. Not shutting down {}. ",
+          server);
+      return 0;
```
Comment (dlmarion) [inline server/manager/src/main/java/org/apache/accumulo/manager/tserverOps/ShutdownTServer.java:64]: I'm not sure that this is doing what the comment suggests. Returning 0 means that the FateOp is ready for execution and the `call` method will be executed.

Comment (dlmarion) [reply server/manager/src/main/java/org/apache/accumulo/manager/tserverOps/ShutdownTServer.java:64]: Nevermind, I see what's happening here. Might want to adjust the comment to say that by returning early the `manager.shutdownTServer` call is never made and the tserver will not be shut down.

Comment (milleruntime) [reply server/manager/src/main/java/org/apache/accumulo/manager/tserverOps/ShutdownTServer.java:64]: Yeah I think we want to throw an error instead, to break out of the operation.

Comment (milleruntime) [reply server/manager/src/main/java/org/apache/accumulo/manager/tserverOps/ShutdownTServer.java:64]: So it works like this but the client thinks that it succeeded. Not sure what happens to the FATE TX. I am going to test it with an error being thrown.

**Comment by milleruntime** [comment]
With an ERROR being thrown instead of logged, it looks like the FATE Tx will actually fail:
<pre>
2022-05-27T12:43:33,843 [fate.Fate] WARN : Failed to execute Repo, FATE[61940a2dc650bf6c]
java.lang.IllegalStateException: Only 1 tserver running and it contains the Root table. Not shutting down localhost:9997[100019a892c0005]
        at org.apache.accumulo.manager.tserverOps.ShutdownTServer.isReady(ShutdownTServer.java:62) ~[accumulo-manager-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at org.apache.accumulo.manager.tserverOps.ShutdownTServer.isReady(ShutdownTServer.java:38) ~[accumulo-manager-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at org.apache.accumulo.manager.tableOps.TraceRepo.isReady(TraceRepo.java:47) ~[accumulo-manager-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at org.apache.accumulo.fate.Fate$TransactionRunner.run(Fate.java:89) ~[accumulo-core-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at org.apache.accumulo.core.trace.TraceWrappedRunnable.run(TraceWrappedRunnable.java:52) ~[accumulo-core-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[?:?]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[?:?]
        at org.apache.accumulo.core.trace.TraceWrappedRunnable.run(TraceWrappedRunnable.java:52) ~[accumulo-core-2.1.0-SNAPSHOT.jar:2.1.0-SNAPSHOT]
        at java.lang.Thread.run(Thread.java:829) ~[?:?]
</pre>

**Comment by milleruntime** [comment]
I don't think this is the right place to do this. I think it would be better to just check in `Admin` before executing the operation. Closing this.

**Pull request closed without merging**